custom/icds/views/ccz_hosting.py

```diff
@@ -0,0 +1,267 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from django.http import HttpResponseRedirect
+from django.utils.translation import (
+    ugettext_lazy,
+    ugettext_noop,
+    ugettext as _,
+)
+from django.utils.functional import cached_property
+from django.utils.decorators import method_decorator
+from django.urls import reverse
+from django.shortcuts import redirect
+from django.views.generic import TemplateView
+from django.contrib import messages
+from django.http import HttpResponse
+from django.conf import settings
+
+from couchexport.models import Format
+from corehq import toggles
+from corehq.apps.domain.auth import get_username_and_password_from_request
+from corehq.apps.app_manager.dbaccessors import (
+    get_brief_apps_in_domain,
+    get_build_by_version,
+)
+from corehq.apps.domain.views import (
+    BaseDomainView,
+    DomainViewMixin,
+)
+from corehq.apps.hqwebapp.decorators import use_select2_v4
+from corehq.apps.locations.permissions import location_safe
+from corehq.apps.domain.decorators import login_and_domain_required
```

Comment:
`login_and_domain_required` seems insufficient for these views - shouldn't it check that the user is a domain admin, say?

Reply:
i would leave that to GS feedback. there was a concern raised when the FF was just at domain level, to which i changed it to allow at user level as well.
Domain admin is a stronger check but from past experience the team has asked for special permission to just limited users configured instead of giving it to all admins.
So giving user based access on FF looked like the quick fix without introducing more complexity.

Reply:
Does that mean dropping the domain level flag too?  With the domain flag turned on, any user on the domain will be able to access these views.  I could also see this switching to "either domain admin or feature flag", but that feels more complicated to implement unless necessary.